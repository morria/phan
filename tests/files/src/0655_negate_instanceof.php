<?php

namespace NS655;

class TestClass {
}
/**
 * @param ?TestClass|?array<int,TestClass> $value
 */
function elseifInfer($value) {
    if ($value instanceof TestClass) {
        $value = [$value];
    } else {
        echo strlen($value);
    }
    foreach ($value ?? [] as $v) {
        echo strlen($v);
    }
}
