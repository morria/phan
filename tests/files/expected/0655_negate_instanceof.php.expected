%s:14 PhanTypeMismatchArgumentInternal Argument 1 (string) is ?array<int,\NS655\TestClass>|array<int,\NS655\TestClass> but \strlen() takes string
%s:17 PhanTypeMismatchArgumentInternal Argument 1 (string) is \NS655\TestClass but \strlen() takes string
